src/atomate2/abinit/sets/base.py

```diff
@@ -0,0 +1,564 @@
+"""Module defining base abinit input set and generator."""
+import copy
+import json
+import logging
+import os
+from collections import namedtuple
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any, ClassVar, Iterable, List, Optional, Union
+
+import pseudo_dojo
+from abipy.abio.input_tags import ION_RELAX, IONCELL_RELAX, MOLECULAR_DYNAMICS, RELAX
+from abipy.abio.inputs import AbinitInput
+from abipy.electrons.gsr import GsrFile
+from abipy.flowtk.utils import Directory, irdvars_for_ext
+from monty.io import zopen
+from monty.json import MSONable
+from pymatgen.core.structure import Structure
+from pymatgen.io.abinit.pseudos import PseudoTable
+from pymatgen.io.core import InputGenerator, InputSet
+
+from atomate2.abinit.files import (
+    fname2ext,
+    load_abinit_input,
+    load_generator,
+    out_to_in,
+)
+from atomate2.abinit.utils.common import (
+    INDATA_PREFIX,
+    INDATAFILE_PREFIX,
+    INDIR_NAME,
+    INPUT_FILE_NAME,
+    OUTDATA_PREFIX,
+    OUTDATAFILE_PREFIX,
+    OUTDIR_NAME,
+    TMPDATA_PREFIX,
+    TMPDIR_NAME,
+    InitializationError,
+)
+
+__all__ = ["AbinitInputSet", "AbinitInputSetGenerator"]
+
+logger = logging.getLogger(__name__)
+
+
+class AbinitInputSet(InputSet):
+    """
+    A class to represent a set of Abinit inputs.
+
+    Parameters
+    ----------
+    abinit_input
+        An AbinitInput object.
+    """
+
+    def __init__(
+        self,
+        abinit_input: AbinitInput,
+        input_files: Optional[Iterable[Union[str, Path, dict]]] = None,
+        link_files: bool = True,
+        validation: bool = True,
+        generator: Optional[MSONable] = None,
+    ):
+        self.input_files = input_files
+        self.link_files = link_files
+        self.validation = validation
+        self.generator = generator
+        super().__init__(inputs={INPUT_FILE_NAME: abinit_input})
+
+    def write_input(
+        self,
+        directory: Union[str, Path],
+        make_dir: bool = True,
+        overwrite: bool = True,
+        zip_inputs: bool = False,
+    ):
+        """Write Abinit input files to a directory."""
+        # TODO: do we allow zipping ? not sure if it really makes sense for abinit as
+        #  the abinit input set also sets up links to previous files, sets up the
+        #  indir, outdir and tmpdir, ...
+        self.inputs["abinit_input.json"] = json.dumps(self.abinit_input.as_dict())
```

Comment:
For the abinit_input_set_generator.json, I've removed it in the end (not needed anymore now as we removed the feature to restart from the previous). For the abinit_input.json, it was just to avoid replicating the abinit_input twice with two different formats somehow. Basically, I could do this in the __init__:

```
def __init__(
        self,
        abinit_input: AbinitInput,
        input_files: Optional[Iterable[Union[str, Path, dict]]] = None,
        link_files: bool = True,
        validation: bool = True,
    ):
        self.input_files = input_files
        self.link_files = link_files
        self.validation = validation
        super().__init__(inputs={INPUT_FILE_NAME: abinit_input,
                         "abinit_input.json": json.dumps(self.abinit_input.as_dict())})
```

What do you think ?

Reply:
I think it is fine, but the plain `json.dumps` might not be safe to get a proper conversion in general. It is probably not neeeded for the AbinitInput, but it may be safer to use 
```python
"abinit_input.json": json.dumps(self.abinit_input.as_dict(), cls=MontyEncoder)
```